File: 0926-flip-string-to-monotone-increasing/0926-flip-string-to-monotone-increasing.cpp
class Solution {
public:
    int minFlipsMonoIncr(string s) {
        int cnt0 = 0, cntf = 0;
        for(auto x : s) {
            if (x == '0') cnt0++;
            else cntf++;
            cnt0 = min(cnt0, cntf);
        }
        return cnt0;
    }
};

File: 0926-flip-string-to-monotone-increasing/0926-flip-string-to-monotone-increasing.rs
impl Solution {
    pub fn min_flips_mono_incr(s: String) -> i32 {
        let (mut cnt1, mut cntf) = (0, 0);
        for x in s.chars() {
            if x == '1' {
                cnt1 += 1;
            } else {
                cntf += 1;
            }
            cntf = cntf.min(cnt1);
        }
        cntf
    }
}